custom/icds_reports/utils/aggregation_helpers/distributed/agg_dashboard_activity.py

```diff
@@ -0,0 +1,276 @@
+from datetime import timedelta
+from django.db.models import Max
+
+from custom.icds_reports.utils.aggregation_helpers.distributed.base import BaseICDSAggregationDistributedHelper
+from custom.icds_reports.const import  AGG_DASHBOARD_ACTIVITY
+from django.utils.functional import cached_property
+from django.contrib.auth.models import User
+from corehq.apps.users.dbaccessors.all_commcare_users import get_user_docs_by_username
+from dimagi.utils.chunked import chunked
+
+
+class DashboardActivityReportAggregate(BaseICDSAggregationDistributedHelper):
+    aggregate_parent_table = AGG_DASHBOARD_ACTIVITY
+
+    def __init__(self, date):
+        self.date = date
+        self.last_agg_date = self.get_last_agg_date()
+
+    def aggregate(self, cursor):
+        drop_query = self.drop_table_query()
+        create_table_query, create_table_param = self.create_table_query()
+        add_query, add_params = self.add_missing_users()
+
+        rollover_query, rollover_param = self.rollover_previous_data()
+        update_queries = self.update_queries()
+
+        cursor.execute(drop_query)
+        cursor.execute(create_table_query, create_table_param)
+        cursor.cursor.execute(add_query, add_params)
+        cursor.execute(rollover_query, rollover_param)
+
+        for query, param in update_queries:
+            cursor.execute(query, param)
+
+    @cached_property
+    def get_dashboard_users(self):
+        dashboard_users = User.objects.filter(username__regex=r'^\d*\.[a-zA-Z]*@.*').all()
+        usernames = {user.username for user in dashboard_users if user.username is not None}
```

Comment:
why is this checking that `user.username is not None`. Because the filter is on `username`, this should already be the case for all `dashboard_users`.

I think that these two lines could be changed to:

`usernames = User.objects.filter(username__regex=...).values_list('username', flat=True)`

Reply:
ah! ok.